common/src/float_ops.rs

```diff
@@ -83,6 +83,23 @@ pub fn is_integer(v: f64) -> bool {
     (v - v.round()).abs() < std::f64::EPSILON
 }
 
+// Formats floats into Python style exponent notation, by first formatting in Rust style
+// exponent notation (`1.0000e0`), then convert to Python style (`1.0000e+00`).
+pub fn format_float_as_exponent(precision: usize, magnitude: f64) -> String {
```

Comment:
Maybe you could have a `upper: bool` param, so you don't have to do `to_lowercase`/`to_uppercase` on the result?

Reply:
I think this change causes unnessessary cost for to_lowercase and to_uppercase for common finite numbers which maybe coolreader18 wanted to point out.
At the same time, I think bool paramter doesn't fit for function design because the parameter need to reperesent upper and lower rather than a bool value.
maybe we need `FormatType::Exponent::Upper` and `FormatType::Exponent::Lower` so that we just can pass the value from `parse` function.